### ISSUE TITLE
Album art loading scaled to 50%.

### ISSUE THREAD

**Issue by Clewsy** [SoongNoonien/mpdevil#84]
Hello friend, I am a big fan of mpdevil.  Love your work.

For a while now, the album art has been loading scaled at what looks like 50% within the window.  Both mini player and the standard view.

I've installed with flatpak and it's up-to-date (version 1.10.3).  I'm running ubuntu 23.10 and was having the same issue on 23.4.  I'm currently using Xorg.  Let me know if you'd like any other details.

![Screenshot from 2023-11-13 08-18-46](https://github.com/SoongNoonien/mpdevil/assets/10361831/e568f01c-d1c8-4742-a826-2ff1ac0a11d3)
![Screenshot from 2023-11-13 08-19-12](https://github.com/SoongNoonien/mpdevil/assets/10361831/f531845e-f75a-45d7-a932-2150827fb561)

**Comment by Clewsy** [comment]
FYI, I just started a new session under wayland and no longer have this issue.  Album art loads correctly.

**Comment by SoongNoonien** [comment]
Hi! Are you using the UI scaling of gtk?

**Comment by Clewsy** [comment]
Yes!  Okay a few quick tests confirm it's not a Wayland/Xorg issue.  I did not have scaling on when I logged into wayland.  After UI scaling to 150% and restarting mpdevil I saw the same album art scaling issue.

The issue is not present in either Wayland or Xorg if I reset UI scaling to 100%.

**Comment by SoongNoonien** [comment]
Commit e9ad0b39f93de942922ab3d29222131a5683a30c should fix this issue. Can you confirm?

**Comment by Clewsy** [comment]
So I removed the flatpak and just tried building from source with the new commit using the guidance in your readme.

```shell
$ meson setup builddir --prefix=/usr/local
The Meson build system
Version: 1.2.1
Source dir: /tmp/mpdevil
Build dir: /tmp/mpdevil/builddir
Build type: native build
Project name: mpdevil
Project version: 1.10.3
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program msgfmt found: NO

data/meson.build:1:5: ERROR: Program 'msgfmt' not found or not executable

A full log can be found at /tmp/mpdevil/builddir/meson-logs/meson-log.txt
```

The log just confirms the same error:
```shelll
Build started at 2023-11-13T10:54:57.492925
Main binary: /usr/bin/python3
Build Options: -Dprefix=/usr/local
Python system: Linux
The Meson build system
Version: 1.2.1
Source dir: /tmp/mpdevil
Build dir: /tmp/mpdevil/builddir
Build type: native build
Project name: mpdevil
Project version: 1.10.3
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86_64
Host machine cpu: x86_64
Target machine cpu family: x86_64
Target machine cpu: x86_64
Program msgfmt found: NO

data/meson.build:1:5: ERROR: Program 'msgfmt' not found or not **executable**
```

I'm not familiar with meson, you might need to give me some additional guidance. :)

**Comment by SoongNoonien** [comment]
It seems like `gettext` is not installed on your system. `sudo apt install gettext` should fix this on ubuntu.

**Comment by Clewsy** [comment]
Next error was:
```shell
Program glib-compile-resources found: NO
```

I had to install ```libglib2.0-dev-bin```.  Then the meson command worked.

**Confirmed, the commit fixed my issue - the album art image is no longer scaled when I use UI scaling.**

**Comment by SoongNoonien** [comment]
> Next error was:

I'll try to fix the build dependencies in the readme.

> Confirmed, the commit fixed my issue - the album art image is no longer scaled when I use UI scaling.

Ok, that is good. But I think it breaks wayland compatibility, is that the case on your machine as well?

**Comment by SoongNoonien** [comment]
> After UI scaling to 150% and restarting mpdevil I saw the same album art scaling issue.

Even under wayland? I tested this over here and it seems like the issue only occurs under xorg in combination with UI scaling.

**Comment by SoongNoonien** [comment]
I've looked into it a little deeper and at least over here commit ed899a13e003888e7bc3494a523b8ac00b7190dd fixes the wayland incompatibility. Can you confirm that there are no regressions compared to e9ad0b39f93de942922ab3d29222131a5683a30c? Also I've updated the readme to improve the build dependencies.

**Comment by Clewsy** [comment]
Hmm after rebuilding and installing from your latest, mpdevil no longer starts for me.  Neither xorg nor wayland.

**Comment by SoongNoonien** [comment]
Is there any output on the console?

**Comment by Clewsy** [comment]
```shell
error: app/org.mpdevil.mpdevil/x86_64/master not installed
```

**Comment by Clewsy** [comment]
What am I doing wrong?

**Comment by SoongNoonien** [comment]
I'm not sure. The message looks like a flatpak error. But if I understood you correctly you removed the flatpak for testing. If you installed mpdevil to `/usr/local` then a simple `/usr/local/bin/mpdevil` should start the program.

**Comment by Clewsy** [comment]
Confirmed, I flatpak uninstalled and have been building from source since.  The which command indicates it's still executing from /usr/local/bin/mpdevil:
```shell
$ which mpdevil
/usr/local/bin/mpdevil
```
However if I run the executable from the absolute path per your last, then I get a different result:
```shell
$ /usr/local/bin/mpdevil 
Traceback (most recent call last):
  File "/usr/local/bin/mpdevil", line 23, in <module>
    from mpd import MPDClient, CommandError, ConnectionError
ModuleNotFoundError: No module named 'mpd'
```

**Comment by Clewsy** [comment]
Okay, I apt installed python3-mpd

**Comment by Clewsy** [comment]
It's running now.  I'll test things and get back to you.

(Not sure why it built before!?  Maybe something I did purged python3-mpd)

**Comment by SoongNoonien** [comment]
> Okay, I apt installed python3-mpd

Yes, exactly. You were missing the mpd module.

> (Not sure why it built before!? Maybe something I did purged python3-mpd)

It should build fine without it. But I'm not sure how you managed to run it the last time.

**Comment by SoongNoonien** [comment]
> The which command indicates it's still executing from /usr/local/bin/mpdevil:

Weird, may I ask what you executed to get the flatpak error before?

**Comment by Clewsy** [comment]
Just ran ```mpdevil``` at the command line.
```shell
$ mpdevil
error: app/org.mpdevil.mpdevil/x86_64/master not installed
```

**Comment by Clewsy** [comment]
Testing your latest was all successful.  Expected album art image sizes (both mini player and standard) under scenarios:

Wayland:
- [x] No UI scaling (100%)
- [x] UI scaling (150%)

xorg:
- [x] No UI scaling (100%)
- [x] UI scaling (150%)

**Comment by SoongNoonien** [comment]
Ok, good. I'll try to make a new release soon.

**Comment by Clewsy** [comment]
Maybe when ```/usr/local/bin/mpdevil``` failed, something caused it to default to some remnant of  my fltapak install?

**Comment by SoongNoonien** [comment]
> Maybe when /usr/local/bin/mpdevil failed, something caused it to default to some remnant of my fltapak install?

I'm not sure but it definitely looks like a flatpak issue. After a quick search I found various reports of similar problems. I hope this repairs itself after you reinstall the flatpak. Also flatpak seems to think you what to start the flatpak from the master branch which I'm not even providing (at least as far as I know).
